pingora-memory-cache/src/read_through.rs

```diff
@@ -181,6 +181,15 @@ where
             }
             None => {
                 let mut lockers = self.lockers.write();
+                // Acquiring write locks is not that easy in a high-concurrency environment,
+                // perhaps another one has already completed the lookup and removed the lock from locker.
+                // So we first check whether data already exists at the current point
+                let (result, cache_state) = self.inner.get(key);
+                if let Some(result) = result {
+                    /* cache hit */
+                    return (Ok(result), cache_state);
+                }
+
```

Comment:
If I understand correct, this is the case following ```match lockers.get(&hashed_key)``` code logic can handle, if another one already added a same ```hashed_key``` just before ```lockers.write()```.

Reply:
suppose there are two threads(coroutine) t1 t2
t1 get data -> None
t1 get container read lock, get CacheLock -> None
t1 get container write lock, insert CacheLock
t1 lookup data
t2 get data -> None
t2 pause for some reason such as thread switching
t1 put lookuped data
t1 get container write lock, remove CacheLock
t2 resume and get container read lock, get CacheLock -> None
t2 get container write lock, insert another CacheLock
t2 lookup data

So it is  waste that t2 lookup data in this situation.
Double check should be done when get  write lock like thread-safe lazy initialization.